### PR TITLE
*: fix some missing set kv variables

### PR DESCRIPTION
**Pull request by tina77fritz**
Signed-off-by: Tina Fritz <tina77fritz@gmail.com>

### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:
changeInvalidToValid will directly active a transaction, so it will have no chance to set kv variables for this kind of transactions. e.g. auto committed point get transaction and auto committed DDL.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:
Add the missing setVars.

How it Works:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

- No release note

### CODE REVIEW

**Comment by tina77fritz** [comment]
PTAL @coocood @lysu

File: session/session.go [hunk 1]
```diff
@@ -1527,6 +1527,7 @@ func (s *session) NewTxn(ctx context.Context) error {
 		StartTS:       txn.StartTS(),
 		ShardStep:     int(s.sessionVars.ShardAllocateStep),
 	}
+	s.txn.SetVars(s.sessionVars.KVVars)
```
Comment (coocood) [inline session/session.go:1530]: This is set already at line 1517

Comment (tina77fritz) [reply session/session.go:1530]: Ah, yes, you are right.

**Comment by tina77fritz** [comment]
PTAL @coocood

**Comment by coocood** [comment]
LGTM

File: store/tikv/txn.go [hunk 2]
```diff
@@ -109,9 +109,17 @@ func (a assertionPair) String() string {
 	return fmt.Sprintf("key: %s, assertion type: %d", a.key, a.assertion)
```
Comment (jackysp) [inline store/tikv/txn.go:109]: Please do not change the file permission.

**Pull request merged**